src/library-authoring/collections/LibraryCollectionPage.tsx

```diff
@@ -92,24 +91,29 @@ const SubHeaderTitle = ({
   );
 };
 
-const LibraryCollectionPageInner = ({ libraryId }: { libraryId: string }) => {
+interface LibraryCollectionPageInnerProps {
+  libraryId: string;
+  collectionId: string;
+}
+
+const LibraryCollectionPageInner = ({ libraryId, collectionId }: LibraryCollectionPageInnerProps) => {
   const intl = useIntl();
 
   const {
     sidebarBodyComponent,
     openCollectionInfoSidebar,
   } = useContext(LibraryContext);
-  const { collectionHits: [collectionData], isFetching } = useSearchContext();
+  const { data: collectionData, isLoading } = useCollection(libraryId, collectionId);
 
   useEffect(() => {
-    openCollectionInfoSidebar();
-  }, []);
+    openCollectionInfoSidebar(collectionId);
+  }, [collectionData]);
 
   const { data: libraryData, isLoading: isLibLoading } = useContentLibrary(libraryId);
 
   // Only show loading if collection data is not fetched from index yet
   // Loading info for search results will be handled by LibraryCollectionComponents component.
-  if (isLibLoading || (!collectionData && isFetching)) {
+  if (isLibLoading || (!collectionData && isLoading)) {
```

Comment:
```suggestion
  if (isLibLoading || isLoading) {
```

`!collectionData` check was only required when fetching both collection metadata and components in a single search call.